code/modules/antagonists/superheroes/buzzon/buzzon_equipment.dm

```diff
@@ -0,0 +1,294 @@
+
+/obj/item/clothing/head/hooded/bee_hood/full/dropped(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/buzzon = user
+
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	hud.remove_hud_from(buzzon)
+	REMOVE_TRAIT(buzzon, TRAIT_DIAGNOSTIC_HUD, CLOTHING_TRAIT)
+	REMOVE_TRAIT(buzzon, TRAIT_ROBOTIC_FRIEND, CLOTHING_TRAIT)
+
+/obj/item/melee/beesword/buzzon
+	name = "The Stinger"
+	desc = "A sting of giant bee on a handle that became the fastest sword in the world thanks to high-tech components."
+	icon_state = "buzzon_sword"
+	inhand_icon_state = "buzzon"
+	worn_icon_state = "buzzon"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	slot_flags = ITEM_SLOT_BACK
+	block_chance = 10 //It's about agility, not about sheer force
+	///If this var is higher than world.time this means that the owner is still in the blocking stance and has a high chance to block the attack
+	var/parry_time = 0
+
+/obj/item/melee/beesword/buzzon/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(parry_time >= world.time)
+		parry_time = 0
+		final_block_chance += 85
+
+	switch(attack_type)
+		if(UNARMED_ATTACK)
+			final_block_chance += 30
+
+		if(PROJECTILE_ATTACK)
+			final_block_chance -= 40
+
+		if(LEAP_ATTACK)
+			if(isliving(hitby))
+				var/mob/living/victim = hitby
+				force = 20
+				attack_verb_continuous = list("penetrates", "pierces") //We pierce them in the flight.
+				attack_verb_simple = list("penetrate", "pierce")
+				attack(victim, owner)
+				force = initial(force)
+				attack_verb_continuous = initial(attack_verb_continuous)
+				attack_verb_simple = initial(attack_verb_simple)
+	. = ..()
+
+/obj/item/melee/beesword/buzzon/attack_secondary(mob/living/victim, mob/living/user, params)
+	user.changeNext_move(CLICK_CD_RANGE)
+	parry_time = world.time + CLICK_CD_RANGE
+	user.visible_message("<span class='warning'>[user] assumes blocking stance.</span>", "<span class='notice'>You assume blocking stance.</span>")
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/melee/beesword/buzzon/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(ishuman(hit_atom))
+		var/mob/living/carbon/human/victim = hit_atom
+		if(istype(victim.get_item_by_slot(ITEM_SLOT_OCLOTHING), /obj/item/clothing/suit/hooded/bee_costume/buzzon))
+			var/obj/item/clothing/suit/hooded/bee_costume/buzzon/suit = victim.get_item_by_slot(ITEM_SLOT_OCLOTHING)
+			if(suit.linked_sword == src)
+				victim.put_in_active_hand(src)
+				victim.visible_message("<span class='warning'>[victim] catches [src] out of the air!</span>")
+				return
+	. = ..()
+
+/mob/living/simple_animal/hostile/bee/toxin_type
+	var/toxin_type = /datum/reagent/toxin
+
+/mob/living/simple_animal/hostile/bee/toxin_type/Initialize()
```

Comment:
you've tested this to make sure that it works with the bee update, right?

Reply:
Yes it does work